### PR TITLE
Changes to pull configuration from GCE metadata for master component …

### PR DESCRIPTION
**Pull request by Q-Lee**
…monitoring.

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/contrib/1663)

### CODE REVIEW

**Comment by Q-Lee** [comment]
How do I re-test in contrib?

File: kubelet-to-gcm/monitor/main/daemon.go [hunk 1]
```diff
@@ -87,10 +87,10 @@ func getZone(zone string) (string, error) {
 	return zone, nil
 }
 
-// getProjectID returns projectID if it's set, and hits the GCE MD server otherwise.
+// getProjectID returns projectID if it's given, or gets it from gce if asked.
 func getProjectID(projectID string) (string, error) {
-	if projectID == "" {
-		body, err := getGCEMetaData("http://169.254.169.254/computeMetadata/v1/project/project-id")
+	if projectID == "use-gce" {
+		body, err := getGCEMetaData("http://169.254.169.254/computeMetadata/v1/project/numeric-project-id")
```
Comment (gmarek) [inline kubelet-to-gcm/monitor/main/daemon.go]: As it is reused multiple times - can you extract the metadata server address prefix as a constant? I.e. const `metadataServerAddress = "http://169.254.169.254/computeMetadata/"` and then `getGCEMetaData(metadataServerAddress + "v1/project/..."`?

File: kubelet-to-gcm/monitor/main/daemon.go [hunk 2]
```diff
@@ -99,6 +99,30 @@ func getProjectID(projectID string) (string, error) {
...
+		if err != nil {
+			return "", fmt.Errorf("Failed to get cluster name from GCE: %v", err)
+		}
+		cluster = string(body)
+	}
+	return cluster, nil
+}
+
+// getKubeletHost returns the kubelet host if given, or gets ip of network interface 0 from gce.
+func getKubeletHost(kubeletHost string) (string, error) {
```
Comment (gmarek) [inline kubelet-to-gcm/monitor/main/daemon.go]: getKubeletHostIP - originally I though it'll return host name.

**Comment by gmarek** [comment]
Generally looks good - just nits. You can apply LGTM yourself when you fix them.

**Comment by eparis** [comment]
If you have `push` permission in contrib you can go to the travis web ui and re-run the test. If not you can push a new commit which will trigger a re-test. You can also close the pr and re-open it to trigger a retest.

**Pull request merged**